Drop after devel/glib20 is updated to 2.58 or later.

meson.build:107:0: ERROR: Invalid version of dependency, need 'gio-2.0' ['>=2.58'] found '2.56.3'.

../tools/dump-app-list.c:25:29: warning: implicit declaration of function 'G_SOURCE_FUNC' is invalid in C99 [-Wimplicit-function-declaration]
  g_timeout_add_seconds (2, G_SOURCE_FUNC (g_main_loop_quit), loop);
                            ^
ld: error: undefined symbol: G_SOURCE_FUNC
>>> referenced by dump-app-list.c
>>>               tools/dump-app-list.p/dump-app-list.c.o:(main)

--- meson.build.orig	2020-08-03 08:10:28 UTC
+++ meson.build
@@ -27,7 +27,7 @@ configure_file(
   configuration: config_h,
 )
 
-glib_ver = 'GLIB_VERSION_2_58'
+glib_ver = 'GLIB_VERSION_2_56'
 
 add_project_arguments([
   '-DHAVE_CONFIG_H',
@@ -104,9 +104,9 @@ gnome = import('gnome')
 i18n  = import('i18n')
 
 gcr_dep = dependency('gcr-3', version: '>= 3.7.5')
-gio_dep = dependency('gio-2.0', version: '>=2.58')
-gio_unix_dep =  dependency('gio-unix-2.0', version: '>=2.58')
-glib_dep = dependency('glib-2.0', version: '>=2.58')
+gio_dep = dependency('gio-2.0', version: '>=2.56')
+gio_unix_dep =  dependency('gio-unix-2.0', version: '>=2.56')
+glib_dep = dependency('glib-2.0', version: '>=2.56')
 gnome_desktop_dep = dependency('gnome-desktop-3.0', version: '>=3.26')
 gobject_dep =  dependency('gobject-2.0', version: '>=2.50.0')
 gtk_dep = dependency('gtk+-3.0', version: '>=3.22')
--- tools/dump-app-list.c.orig	2020-08-03 08:10:28 UTC
+++ tools/dump-app-list.c
@@ -10,6 +10,10 @@
 
 #include <app-list-model.h>
 
+#if !GLIB_CHECK_VERSION(2, 58, 0)
+#define G_SOURCE_FUNC(f) ((GSourceFunc) (void (*)(void)) (f))
+#endif
+
 int
 main (int argc, char **argv)
 {
